packages/ui-kit/src/components/AccessTokenProvider/AccessTokenProvider.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { createContext, PropsWithChildren, useCallback, useEffect, useRef, useState } from 'react'
+import { useLog } from '../Log'
+
+interface AccessTokenContextValue {
+  accessToken?: string
+  isLoading?: boolean
+  onExpiredToken?: () => void
+  failedRetry?: boolean
+}
+
+const ACCESS_TOKEN_REFRESH_INTERVAL = 3600000
```

Comment:
It may make sense to subtract five minutes from this, to allow for time fetching the replacement token.

```suggestion
const ACCESS_TOKEN_REFRESH_INTERVAL = 3600000
```